include/treelite/tree.h

```diff
@@ -769,31 +783,37 @@ class Model {
    * \brief number of features used for the model.
    * It is assumed that all feature indices are between 0 and [num_feature]-1.
    */
-  int num_feature;
+  int32_t num_feature{0};
   /*! \brief Task type */
   TaskType task_type;
   /*! \brief whether to average tree outputs */
-  bool average_tree_output;
+  bool average_tree_output{false};
   /*! \brief Group of parameters that are specific to the particular task type */
-  TaskParam task_param;
+  TaskParam task_param{};
   /*! \brief extra parameters */
-  ModelParam param;
+  ModelParam param{};
+
+ protected:
+  // Magic numbers, to be computed as part of serialization
+  uint64_t num_tree_{0};
+  int32_t num_opt_field_per_model_{0};
+  int32_t major_ver_, minor_ver_, patch_ver_;
```

Comment:
```suggestion
  int32_t major_ver_;
  int32_t minor_ver_;
  int32_t patch_ver_;
```
Separating each declaration will make it easier for folks to see what's going on at a glance and avoid some non-obvious bugs with accidental invocation of the comma operator or confusing pointer declarations.